controllers/warehouseController.js

```diff
@@ -0,0 +1,58 @@
+const Joi = require("joi");
+const knex = require("knex")(require("../knexfile"));
+
+const updateWarehouseDetails = async (req, res) => {
+  try {
+    const warehouseId = parseInt(req.params.id);
+
+    if (!warehouseId || isNaN(warehouseId)) {
+      return res.status(400).json({ message: "Invalid warehouse ID." });
+    }
+
```

Comment:
If you modify your route to app.use("/warehouses/:id", warehouseRoutes);, then the API endpoint will only be accessible when the request includes a valid warehouse ID, such as warehouses/554421. This eliminates the need to manually check if the ID was passed or if it's invalid in the controller.